src/libs/actions/IOU.js

```diff
@@ -1121,44 +1126,46 @@ function sendMoneyViaPaypal(report, amount, currency, comment, managerEmail, rec
 }
 
 /**
+ * @param {String} paymentType
  * @param {Object} chatReport
  * @param {Object} iouReport
  * @param {Object} recipient
+ * @param {String} reimbursementBankAccountState
  */
-function payMoneyRequestElsewhere(chatReport, iouReport, recipient) {
-    const {params, optimisticData, successData, failureData} = getPayMoneyRequestParams(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.ELSEWHERE);
-
-    API.write('PayMoneyRequestElsewhere', params, {optimisticData, successData, failureData});
-
-    Navigation.navigate(ROUTES.getReportRoute(chatReport.reportID));
-}
-
-/**
- * @param {Object} chatReport
- * @param {Object} iouReport
- * @param {Object} recipient
- */
-function payMoneyRequestWithWallet(chatReport, iouReport, recipient) {
-    const {params, optimisticData, successData, failureData} = getPayMoneyRequestParams(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.EXPENSIFY);
-
-    API.write('PayMoneyRequestWithWallet', params, {optimisticData, successData, failureData});
-
-    Navigation.navigate(ROUTES.getReportRoute(chatReport.reportID));
-}
-
-/**
- * @param {Object} chatReport
- * @param {Object} iouReport
- * @param {Object} recipient
- */
-function payMoneyRequestViaPaypal(chatReport, iouReport, recipient) {
-    const {params, optimisticData, successData, failureData} = getPayMoneyRequestParams(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.PAYPAL_ME);
-
-    API.write('PayMoneyRequestViaPaypal', params, {optimisticData, successData, failureData});
+function payMoneyRequest(paymentType, chatReport, iouReport, reimbursementBankAccountState) {
+    if (
+        chatReport.chatType === CONST.REPORT.CHAT_TYPE.POLICY_EXPENSE_CHAT &&
+        reimbursementBankAccountState !== CONST.BANK_ACCOUNT.STATE.OPEN &&
+        paymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY
+    ) {
+        Navigation.navigate(ROUTES.BANK_ACCOUNT_NEW);
+        return;
+    }
```

Comment:
I think this is not needed since it is already handled by KYCWall?

Reply:
I think KYCWall might handle this, but we should update the logic in in. For example, [here](https://github.com/Expensify/App/blob/4905557218cf29fc2e925941c035cd844debf555/src/components/KYCWall/BaseKYCWall.js#L111-L117) we ask users to upgrade to a gold wallet, which we don't need to pay a Manual Request. 

Also, we call [hasExpensifyPaymentMethod](https://github.com/Expensify/App/blob/4905557218cf29fc2e925941c035cd844debf555/src/components/KYCWall/BaseKYCWall.js#L100) which seems to check for a [credit account](https://github.com/Expensify/App/blob/4905557218cf29fc2e925941c035cd844debf555/src/libs/PaymentUtils.js#L19) and not withdrawal?

Reply:
hmmm ok, so what do I need to do exactly?